pandas/_testing/__init__.py

```diff
@@ -292,21 +296,40 @@ def makeBoolIndex(k=10, name=None):
     return Index([False, True] + [False] * (k - 2), name=name)
 
 
+def makeNumericIndex(k=10, name=None, *, dtype):
+    dtype = pandas_dtype(dtype)
```

Comment:
do we expect non-numpy dtypes here?

Reply:
I’ve added an assert for that here.

Reply:
my idea was to just use `dtype = np.dtype(dtype)`

Reply:
```python
>>> np.dtype("Int8")  # capital Int
DeprecationWarning: Numeric-style type codes are deprecated and will result in an error in the future.
  np.dtype("Int8")
dtype('int8')
```

So I think we should hold on this until the above raises IMO.

Reply:
this will not ever raise as these are pandas drypes

Reply:
@jbrockmendel raises a valid point; if u r expecting only numpy dtypes then why are pandas dtypes passed?

Reply:
> this will not ever raise as these are pandas dtypes

`assert isinstance(dtype, np.dtype)` on line 304 ensures that it will raise if it is a pandas dtype.

I'm not intending to pass pandas dtypes here, but I'm using `pandas_dtype` to avoid an ambiguity: The string `"Int64"` in Pandas means `Int64Dtype`, but in numpy it means `np.int64`. So by using `dtype = pandas_dtype(dtype)`, then an `assert(dtype, np.dtype)`, we ensure the correct dtype..